### ISSUE TITLE
Assert.NotNull and .Should().NotBeNull() aren't treated as a null reference check (Incorrect warning: "CS8602: Dereference of a possibly null reference.")

### ISSUE THREAD

**Issue by Sa1Gur** [dotnet/roslyn#57067]
**Version Used**:  6.0.100-preview.7.21379.14

**Steps to Reproduce**:

1. Create XUnit project net 5.0 and NRT enabled
2. Write code
```
using FluentAssertions;
using Xunit;

namespace TestProject1
{
    public class UnitTest1
    {
        class TestClass
        {
            public string TestProperty { get; set; } = "is_string";
        }

        static TestClass? TestClassReturner() => new TestClass();
        
        [Fact]
        public void Test1()
        {
            var test = TestClassReturner();

            Assert.NotNull(test);
            Assert.Equal( "is_string",test.TestProperty);
        }
    
        [Fact]
        public void Test2()
        {
            var test = TestClassReturner();

            test.Should().NotBeNull();
            test.TestProperty.Should().Be("is_string");
        }
    }
}
```
3. 
**Expected Behavior**: No warning
**Actual Behavior**: Compiler warning: "CS8602: Dereference of a possibly null reference."

**Comment by Sa1Gur** [comment]
Unfortunately the first test behavior is the same as the second one. I've got warnings in both

**Comment by Sa1Gur** [comment]
In the 2.4.2-pre.12 on NuGet: https://www.nuget.org/packages/xunit.assert/2.4.2-pre.12 issue with Assert is resolved.

https://github.com/xunit/xunit/discussions/2414#discussioncomment-1600003
